src/pybind/mgr/dashboard/frontend/cypress/integration/cluster/create-cluster.po.ts

```diff
@@ -127,4 +131,61 @@ export class CreateClusterWizardHelper extends PageHelper {
       cy.get('@addButton').click();
     });
   }
+
+  private selectServiceType(serviceType: string) {
+    return this.selectOption('service_type', serviceType);
+  }
+
+  addService(serviceType: string) {
+    cy.wait(30000);
```

Comment:
So the new e2e test will be blindly waiting for things to happen during 2.5 minutes (30 s + 30 s + 90 s). Can't we just make our tests conditionally wait for an event? That is: regularly checking whether some condition happened (service was created, service was deleted) up to max 30 seconds (if you use an conditional wait, you can permit yourself even higher wait times, 60 secs, under the assumption that the test in average will pass quickly).

This kind of explicit waits are counterproductive, they're not really effective since you have to overestimate them to deal with different HW setups. So this is why e2e usually is so flaky and prone to false errors (and why people like me try to avoid e2e testing vs. unit/component testing)

Reply:
I added these waits for the tests to pass in my local cluster(which is running very slow). Will remove them from here.